Jenkinsfile

```diff
@@ -95,18 +95,25 @@ helpers.rootLinuxNode(env, {
             }
         }
 
+        def hasGoChanges = hasChanges('go')
+        def hasJSChanges = hasChanges('shared')
```

Comment:
Yeah I haven't actually moved any of the JS checking yet, this is just output for now. The JS script has more functionality that isn't as critical to implement in the Jenkinsfile as we don't need its reuse.